sdk/core/azure-core/tests/async_tests/test_authentication_async.py

```diff
@@ -226,3 +228,125 @@ def get_completed_future(result=None):
     fut = asyncio.Future()
     fut.set_result(result)
     return fut
+
+@pytest.mark.asyncio
+async def test_bearer_policy_redirect_same_domain():
+    class MockTransport(AsyncHttpTransport):
+        def __init__(self):
+            self._first = True
+        async def __aexit__(self, exc_type, exc_val, exc_tb):
+            pass
+        async def close(self):
+            pass
+        async def open(self):
+            pass
+
+        async def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineResponse
```

Comment:
can we use the testserver to mock tests?

Reply:
I am afraid testserver cannot mock the scenario that new target is in different domain.